Engine/src/Engine/Audio.cpp

```diff
@@ -19,38 +21,114 @@ AudioPlayerSingleton* AudioPlayerSingleton::getInstance()
 // Plays a sound, takes a path to the sound as a const char, a bool if the sound should loop,
 // a bool if the sound should be initialized paused or play on initialization, and a bool for if
 // sound effects should be used. Sound effects need to be added after using this function using
-// the returned ISound*.
+// the returned ISound*.  If the sound cannot be found, it returns a nullptr.
+// If useFullPath is true, the full path to the sound file can be used. Otherwise, only the sound name
+// needs to be put in with the file type. (E.g.:"GameEngine/Engine/src/Media/sound.wav" vs "sound.wav")
 // More information on sound effects: https://www.ambiera.com/irrklang/docu/index.html#effects
 // 
-ISound* AudioPlayerSingleton::playSound(const char* soundPath, bool loop, bool startPaused, bool useSoundEffects)
+ISound* AudioPlayerSingleton::playSound(const char* soundNamePath, bool loop, bool startPaused, bool useSoundEffects, bool useFullPath)
 {
-	ISound* theSound = engine->play2D(soundPath, loop, startPaused, true, ESM_AUTO_DETECT, useSoundEffects);
-	if (theSound != 0)
+	if (useFullPath)
 	{
-		return theSound;
+		ISound* theSound = engine->play2D(soundNamePath, loop, startPaused, true, ESM_AUTO_DETECT, useSoundEffects);
+		if (theSound != 0)
+		{
+			return theSound;
+		}
+		else
+		{
+			return nullptr;
+		}
+	}
+	else
+	{
+		string filePath = std::filesystem::current_path().parent_path().string() + "\\Engine\\src\\Media\\";
+		filePath += soundNamePath;
+		std::replace(filePath.begin(), filePath.end(), '\\', '/');
+		const char* fullPath = filePath.c_str();
+		ISound* theSound = engine->play2D(fullPath, loop, startPaused, true, ESM_AUTO_DETECT, useSoundEffects);
+		if (theSound != 0)
+		{
+			return theSound;
+		}
+		else
+		{
+			string filePath2 = std::filesystem::current_path().parent_path().string() + "\\Game\\src\\Media\\";
+			for (const auto& dEntry : std::filesystem::recursive_directory_iterator(filePath2))
+			{
+				if (dEntry.path().filename() == soundNamePath)
+				{
+					ISound* theSound2 = engine->play2D(dEntry.path().string().c_str(), loop, startPaused, true, ESM_AUTO_DETECT, useSoundEffects);
```

Comment:
This line might be a bit too long, should be shortened according to standards
(Could just show up that way though since it's on github web that looks long. Just double check in IDE)